src/pgduckdb_background_worker.cpp

```diff
@@ -932,12 +933,12 @@ CreateSchemaIfNotExists(const char *postgres_schema_name, bool is_default_db) {
 		    .objectId = schema_oid,
 		    .objectSubId = 0,
 		};
-		ObjectAddress extension_address = {
-		    .classId = ExtensionRelationId,
-		    .objectId = pgduckdb::ExtensionOid(),
+		ObjectAddress server_address = {
+		    .classId = ForeignServerRelationId,
+		    .objectId = GetMotherduckForeignServerOid(), // pgduckdb::ExtensionOid(),
 		    .objectSubId = 0,
 		};
-		recordDependencyOn(&schema_address, &extension_address, DEPENDENCY_NORMAL);
+		recordDependencyOn(&schema_address, &server_address, DEPENDENCY_NORMAL);
```

Comment:
This works for tables in a `ddb$` schema, because those will then depend on the SERVER indirectly through their schema. But this doesn't yet work for tables in the default database, for those we need to create a direct dependency between the table and the schema. (and probably we just want to create the direct dependency for any `duckdb` table, no matter what schema they are in, just for consistency).